docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -1800,291 +1800,343 @@ The following error message codes may appear on the app-server log. Use the foll
+  Skipping invalid plugin group=_plugins_
 
   **Reason:**
 
-  TODO
+  If _plugins_ is not an array and the size is less than zero, then it will log a warning message.
 
   **Action:**
 
-  TODO
+  Make sure _plugins_ should be an array of size greater than zero.
+
 
 
+### ZWED0077W
 
-### ZWED0077W":"Could not extract war for service=%s, error=
+  Could not extract war for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If the service with the _key-value_ pair is unable to extract the war file then it throws the _errorMessage_
 
   **Action:**
 
-  TODO
+  Check if the war file exists and configured correctly.
 
 
 
-### ZWED0078W":"Could not access files to determine status for service=%s, error=
+### ZWED0078W
+
+  Could not access files to determine status for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If we are unable to get the status of war extracted or not, then it throws _errorMessage_ in catch block.
 
   **Action:**
 
-  TODO
+  Check if the war file exists.
+
 
 
+### ZWED0079W
 
-### ZWED0079W":"Cannot add servlet for service=%s, error=
+  Cannot add servlet for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If unable to add servlet for service _key-value_, then it logs a warning _errorMessage_.
 
   **Action:**
 
-  TODO
+  No action is needed, but the warning may be needed to debug this War
+
 
 
+### ZWED0080W
 
-### ZWED0080W":"Cannot add servlet for service=%s
+  Cannot add servlet for service=_key-value_
 
   **Reason:**
 
-  TODO
+  When we are not able to get the directory to add servlet for service _key-value_.
 
   **Action:**
 
-  TODO
+  Check if your directory is exists and is valid.
 
 
 
-### ZWED0081W":"Could not start tomcat, error=
+### ZWED0081W
+
+  Could not start Tomcat, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  Tomcat manager is unable to start Tomcat with the Java option, due to an incorrect configuration with `components.app-server.node.https.port`, `components.app-server.node.https.key`, or `components.app-server.node.https.certificate`.
 
   **Action:**
 
-  TODO
+  Verify configuration with `components.app-server.node.https.port`, `components.app-server.node.https.key`, or `components.app-server.node.https.certificate` is valid or not.
+
 
 
+### ZWED0082W
 
-### ZWED0082W":"%s stderr=%s
+  Tomcat PID=_pid_: stderr=_error_
 
   **Reason:**
 
-  TODO
+  A Tomcat process with Tomcat _pid_ encountered an _error_ (stderr).
 
   **Action:**
 
-  TODO
+  Action depends on what _error_ is and may be useful to debug.
+
 
 
+### ZWED0083W
 
-### ZWED0083W":"Tomcat could not start. Closing. code=%s
+  Tomcat could not start. Closing. code=_code_
 
   **Reason:**
 
-  TODO
+  If the Tomcat manager is unable to start itself, then it closes with _code_.
 
   **Action:**
 
-  TODO
+  Review the message and if app server service is interrupted, send it and your log to support for troubleshooting.
```

Comment:
If app server service is interrupted, go through the message for details on what to debug or send logs and message to support.